src/base/transport.ts

```diff
@@ -0,0 +1,130 @@
+import OAuth from 'oauth';
+import IClientOptions from './IClientOptions';
+import Cache from './Cache';
+import { formatURL } from './utils';
+
+class Transport {
+  private oauth: OAuth.OAuth;
+
+  private cache?: Cache;
+  private credentials: IClientOptions & { [key: string]: any };
+  constructor(options: IClientOptions) {
+    this.credentials = options;
+    this.oauth = new OAuth.OAuth(
+      'https://api.twitter.com/oauth/request_token',
+      'https://api.twitter.com/oauth/access_token',
+      this.credentials.apiKey,
+      this.credentials.apiSecret,
+      '1.0A',
+      null,
+      'HMAC-SHA1'
+    );
+    if (!options?.disableCache) {
```

Comment:
I would suggest an empty line here as well, before the if statement :slightly_smiling_face:

Reply:
done